src/java.base/share/native/launcher/main.c

```diff
@@ -167,6 +167,49 @@ static void setup_sighandler() {
     }
 }
 
+static const char *last_pid_filename = "/proc/sys/kernel/ns_last_pid";
+
+static int get_last_pid() {
+    FILE *last_pid_file = fopen(last_pid_filename, "r");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
+        return -1;
+    }
+    int last_pid;
+    if (0 >= fscanf(last_pid_file, "%d", &last_pid)) {
+        fclose(last_pid_file);
+        perror("last_pid_file fscanf");
+        last_pid = -1;
+    }
+    fclose(last_pid_file);
+    return last_pid;
+}
+
+static void set_last_pid(int pid) {
+    FILE *last_pid_file = fopen(last_pid_filename, "w");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
```

Comment:
Should we have this message printed, even if we can achieve it through spinning? Looks like a unnecessary noise when we're still 'fine'.

Reply:
In non-privileged containers, ns_last_pid cannot be even open for writing. So this message may be considered as a warning, or info, or suppressed at all as Radim suggested.

Reply:
Oh, of course. In case we cannot even open that, and that is expected, we should not print anything.